engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -341,4 +341,37 @@ - (int64_t)lastAppModificationTime {
   return [fileDate timeIntervalSince1970];
 }
 
+- (UISceneConfiguration*)application:(UIApplication*)application
+    configurationForConnectingSceneSession:(UISceneSession*)connectingSceneSession
+                                   options:(UISceneConnectionOptions*)options {
+  // Not called before iOS 13.
+  if (@available(iOS 13.0, *)) {
+    NSDictionary* sceneManifest =
+        [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIApplicationSceneManifest"];
+    NSDictionary* sceneConfigs = sceneManifest[@"UISceneConfigurations"];
+    NSArray* roleConfigs = sceneConfigs[connectingSceneSession.role];
+
+    if (roleConfigs.count > 0) {
+      // Defer to the Info.plist if it is specified there.
+      return nil;
```

Comment:
Hmmm, from the API the return type isn't optional: https://developer.apple.com/documentation/uikit/uiapplicationdelegate/application(_:configurationforconnecting:options:)

Reply:
The objc signature is ambiguous so out of curiosity I tried returning nil and it just silently fails.  I'll have to rework this.

Reply:
We should return the first config (which is the default config) in the list here, with sanity check of the `roleConfig` key/values. 

Alternatively a better/simpler approach, I *think* you can just do a `super` call, which should do exactly the above for our case (i.e. no multiple scenes support). 

```
return [super configurationForConnect...];
```

Reply:
There is no super implementation, we are getting that selector from a protocol.  Unfortunately it seems like they are using reflection to change the behavior so we can't send it down the default path.

Reply:
> we are getting that selector from a protocol.

oh you are right. Let's try returning the first config. 

> Unfortunately it seems like they are using reflection to change the behavior so we can't send it down the default path.

My understanding is that UIKit just do something like this: 

```
// Somewhere in UIKit
let config = makeTheDefaultConfigFromPlist()
if ([appDelegate respondsToSelector(configurationForConnect...)] {
  config = [appDelegate configurationForConnect...]
}
```

Reply:
I implemented reading from the plist, it wasn't a big deal.